src/core/ext/transport/chttp2/server/chttp2_server.cc

```diff
@@ -170,8 +172,9 @@ class Chttp2ServerListener : public Server::ListenerInterface {
           ABSL_GUARDED_BY(&connection_->mu_);
       // State for enforcing handshake timeout on receiving HTTP/2 settings.
       Timestamp const deadline_;
-      grpc_timer timer_ ABSL_GUARDED_BY(&connection_->mu_);
-      grpc_closure on_timeout_ ABSL_GUARDED_BY(&connection_->mu_);
+      absl::optional<EventEngine::TaskHandle> timer_handle_
+          ABSL_GUARDED_BY(&connection_->mu_);
+      EventEngine* event_engine_ ABSL_GUARDED_BY(&connection_->mu_);
```

Comment:
I don't think we need to store this pointer again in this object.  We can instead use `connection_->event_engine_`.

Reply:
Good point. Removed. I also changed to grab the `ActiveConnection::event_engine_` in its constructor instead of inside `ActionConnection::Start()`. I don't remember why I did that before...